Add Jan update email (#70)
srobo/team-emails: SR2021/2021-01-01-january-update.md

---
to: SR2021 teams
subject: Student Robotics January Update
---

Hello there!

> There is less than 2 weeks until our [first round of league matches](https://studentrobotics.org/events/sr2021/league-1/) so we wanted to check in and give you an update.

Reviewer: grammars
```suggestion
There are less than 2 weeks until our [first round of league matches](https://studentrobotics.org/events/sr2021/league-1/) so we wanted to check in and give you an update.
```

## League Sessions

> The first league session will be happening on the 16th of January, at 12:00 UK time, lasting around 1.5 hours. You will need to have submitted your code to us (more on that below) by 09:00 UK time on the same day. If you do not submit code, or submit late, you will be marked absent and score 0 points.

Reviewer 2: Only allowing 3hrs between code submission and live-stream seems quite short. How long did we allow last year?

Reviewer: I think we allowed 5 hours -- 10am to 3pm. In practise I think a good portion of that we didn't use though. This year I'd hope we're able to work faster too, given the amount of automation. I agree we should perhaps allow a bit more time for the first of the sessions, in case we need it?

Author: Automation should help us massively, but I agree more time for the first is definitely a good idea. Have dialed back to 08:00.

Reviewer: :+1: to getting us a bit more time. Not sure if this is a silly question -- does 8am really help? Are we going to be awake at that time in the morning?

> The first league session will be happening on the 16th of January, at 12:00 UK time, lasting around 1.5 hours. You will need to have submitted your code to us (more on that below) by 09:00 UK time on the same day. If you do not submit code, or submit late, you will be marked absent and score 0 points.

Reviewer: Have we worked out what time these are in Australia? Even if we're not including them in the email, it might be worth having it on the PR for reference.

Reviewer 2: 09:00Z is 19:30 in Adelade and 12:00Z is 22:30 in Adelade.

Author: Far from the best times, but unfortunately with 10.5h difference, it's the best we could do.

Further details, and a link to the livestream, will be on our website soon - for this and the other league sessions.

## Friendlies

As noted in our previous email, we're going to be running some friendly matches on the 9th of January. These matches are entirely optional, and will not contribute to the league. The matches are simply a chance to see how your robot performs against other robots, as well as socialise with other teams and us volunteers. Unlike our other events, these matches will be incredibly relaxed and informal. We encourage anyone to join in the conversation in Discord or in the chat.

> [Watch our livestream](https://youtu.be/iwa-7R6sR8M).

Reviewer: Do we really want to livestream the friendlies? Not sure if that's what this line means, however it seems counter to "relaxed and informal" to then also be livestreaming them.

Author: It's an unlisted video. YouTube is the easiest way to get the footage out and have a good feedback loop between teams watching live.

Reviewer: Ah, I think possibly I'm misunderstanding the goal here. I had assumed that friendlies were much more akin to Tinker Time -- that we'd probably be giving the teams their logs and the recordings without any fanfare, to allow them to debug stuff having interacted with another robot. Is something else the goal here?

Perhaps it would be useful to clarify what form the livestream will take -- are we expecting to have commentary on it for example? If it's just that we're going to use the livestream to distribute the videos I think we should say that explicitly.

Author: There will be a commentary from any volunteer who's around, much like mentoring at a tech day. There won't be any play-by-play commentary on a match, or at least far from the same format of a regular competition.

My expectation is basically for it to be a livestreamed Meet call with some volunteers, where we show the matches, chat with competitors, comment on performance, ask questions, things like that.

If you are interested in entering our friendly matches, please let us know by either responding to this email, or in `#friendly-matches` on Discord.

## Code Submission

Because the competition is virtual this year, you'll need to send your code to us to run using our code submission tool.

> In the next few days, you'll receive login credentials for this submitter, 1 set for each team. Here you'll be able to submit your code for us to run. The code should be in a zip file, and the entry point for your robot program named `robot.py`.

Reviewer: style nit:
```suggestion
In the next few days, you'll receive login credentials for this submitter, one set for each team. Here you'll be able to submit your code for us to run. The code should be in a zip file, and the entry point for your robot program named `robot.py`.
```

Further details, and a link to the livestream, will be on our website soon - for this and the other league sessions.

## Simulator Update

We released a new version of the simulator, which adds a few important improvements:

- Support for Webots 2021a (no more deprecation warnings).
- Add several scientific libraries. See [the docs](https://studentrobotics.org/docs/programming/python/libraries) for a full list.
- Reconstructed arena to fix issues with distance sensors and shadows. **Note**: the new arena is _slightly different_ now, mostly that the interior (diagonal) walls have moved by a few centimetres.
- The error message about rate-limited LEDs now includes the ruggeduino pin number for the LED, rather than its name.

This can be downloaded from [the docs](https://studentrobotics.org/docs/simulator/#installing-the-simulation). We recommend updating and testing your code with this version as soon as possible. Both the first league session and friendlies will use this version.

## Friendlies

> As noted in our previous email, we're going to be running some friendly matches on the 9th of January. These matches are entirely optional, and will not contribute to the league. The matches are simply a chance to see how your robot performs against other robots, as well as socialise with other teams and us volunteers. Unlike our other events, these matches will be incredibly relaxed and informal. We encourage anyone to join in the conversation in Discord or in the chat.
> 
> [Watch our livestream](https://youtu.be/iwa-7R6sR8M).

Reviewer: Suggest inlining the livestream link? I _think_ this makes it feel a bit less formal, which seems to be the goal of the friendlies.

```suggestion
As noted in our previous email, we're going to be running some friendly matches on the 9th of January. These matches are entirely optional, and will not contribute to the league. The matches are simply a chance to see how your robot performs against other robots, as well as socialise with other teams and us volunteers.

We'll use a YouTube [livestream](https://youtu.be/iwa-7R6sR8M) to broadcast the matches and as usual we'll provide your logs afterwards. Unlike our other events, these matches will be incredibly relaxed and informal. We encourage anyone to join in the conversation in Discord or in the chat.
```

Further details, and a link to the livestream, will be on our website soon - for this and the other league sessions.

## Simulator Update

We released a new version of the simulator, which adds a few important improvements:

- Support for Webots 2021a (no more deprecation warnings).
- Add several scientific libraries. See [the docs](https://studentrobotics.org/docs/programming/python/libraries) for a full list.

> - Reconstructed arena to fix issues with distance sensors and shadows.

Reviewer: I think we should explicitly call out here that as a result of this rebuild the positions of the internal walls have changed slightly. It's only a couple of centimetres at most, but if someone is dead-reckoning it might throw them off.

Reviewer: 92374ce726dac21c2b998af2eaafaa75dbabfb53
